Sources/MapboxNavigation/OrnamentsController.swift

```diff
@@ -252,7 +288,20 @@ extension NavigationMapView {
                     var latestFeature: MBXFeature?
                     let slicedLine = stepShape.sliced(from: closestCoordinate)!
                     
+                    var largestJaccardSimilarity = 0.0
+                    var similarFeature: MBXFeature?
+
                     for queriedFeature in queriedFeatures {
+                        // Calculate the similarity between the EHorizon road name and the feature property road name, and then use the most similar one for the road label.
+                        if let roadName = queriedFeature.feature.properties["name"] as? String,
+                           !roadName.isEmpty {
+                            let jaccardSimilarity = self.calculateJaccardSimilarity(targetString: roadName, compareSet: eHorizonRoadNameSet, allCounts: eHorizonRoadNameCount)
+                            if jaccardSimilarity > largestJaccardSimilarity {
+                                largestJaccardSimilarity = jaccardSimilarity
+                                similarFeature = queriedFeature.feature
+                            }
+                        }
```

Comment:
Do we really need to implement comparison algorithm and choose the one which is the most accurate? What about choosing the one provided by Electronic Horizon by default?

Reply:
Because right now we still need the feature which contains the `shield`  and `reflen`  property from the queried result to add into the `wayNameView` . If we only update the name from the EHorizon, the other properties maybe delayed compared with the name.  Now the comparison is to pick up the best match query result for the EHorizon name and then update the `wayNameView` based on previous logics.